bumble/profiles/bap.py

```diff
@@ -452,6 +759,417 @@ def __init__(
         super().__init__(characteristics)
 
 
+class AseStateMachine(gatt.Characteristic):
```

Comment:
Would it make sense to have the characteristic as a property of `AseStateMachine` rather than directly inheriting from `gatt.Characteristic`? The characteristic value (read and write) can easily be delegated to this object.

Reply:
Maybe in another PR? Currently the design can avoid ASCS main service accessing members of characteristics.